### PR TITLE
Updating migration docs and adding hint about polystat panel

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What this PR does / why we need it**:
Adding migration guide for version 9 for plugin developers and a hit about polystat plugin compatibility with version 9

### CODE REVIEW

File: docs/sources/developers/plugins/migration-guide.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ title: Plugin migration guide
 
 ## Introduction
 
-This guide helps you identify the steps you need to take based on the Grafana version your plugin supports and explains how to migrate the plugin to the 8.2.x or a later version.
+This guide helps you identify the steps you need to take based on the Grafana version your plugin supports and explains how to migrate the plugin to the 9.x or a later version.
```
Comment (jackw) [inline docs/sources/developers/plugins/migration-guide.md:11]: Not sure we should need to keep changing this intro. Maybe we could use something more generic?

```suggestion
The guide helps you identify the steps required to update a plugin from the Grafana version it currently supports to newer versions of Grafana.
```

File: docs/sources/developers/plugins/migration-guide.md [hunk 2]
```diff
@@ -17,6 +17,13 @@ This guide helps you identify the steps you need to take based on the Grafana ve
 - [Plugin migration guide](#plugin-migration-guide)
   - [Introduction](#introduction)
   - [Table of contents](#table-of-contents)
+  - [From version 8.x to 9.x](#from-version-8x-to-9x)
+    - [9.0 deprecations](#90-deprecations)
```
Comment (jackw) [inline docs/sources/developers/plugins/migration-guide.md:21]: Are these deprecations or breaking changes?

File: docs/sources/developers/plugins/migration-guide.md [hunk 3]
```diff
@@ -51,6 +58,66 @@ This guide helps you identify the steps you need to take based on the Grafana ve
...
+```
+
+#### GrafanaTheme2 and useStyles2 replaces getFormStyles
+
+We have removed the deprecated `getFormStyles` function from [grafana-ui](https://www.npmjs.com/package/@grafana/ui). Use `GrafanaTheme2` and the `useStyles2` hook instead
+
+#### /api/ds/query replaces /api/tsdb/query
+
+We have removed the deprecated `/api/tsdb/query` metrics endpoint. Use [/api/ds/query]({{< relref "../http_api/data_source/#query-a-data-source" >}}) instead
+
```
Comment (jackw) [inline docs/sources/developers/plugins/migration-guide.md:120]: Would it be worth adding anything related to the move to webpack 5 in toolkit and possible impact with custom webpack configs / node polyfills? https://grafana.com/docs/grafana/latest/release-notes/release-notes-9-0-0-beta2/#breaking-changes

Maybe a mention of the removal of `selectOptionInTest` from `@grafana/ui` also? I don't think it'll impact anyone but we never know... https://grafana.com/docs/grafana/latest/release-notes/release-notes-9-0-0/#breaking-changes

**Review by jackw — APPROVED** [review]
Awesome work on this @tolzhabayev . I've left a couple of nits. Up to you if think they're worth chasing up on. 🚀

File: docs/sources/developers/plugins/migration-guide.md (hunk 2 above)
Comment (tolzhabayev) [reply docs/sources/developers/plugins/migration-guide.md]: Good point. Changed https://github.com/grafana/grafana/pull/51367/commits/1ce3f43aae0e2d45d7954a25425a2812c414336e

File: docs/sources/developers/plugins/migration-guide.md (hunk 3 above)
Comment (tolzhabayev) [reply docs/sources/developers/plugins/migration-guide.md:122]: Added https://github.com/grafana/grafana/pull/51367/commits/1ce3f43aae0e2d45d7954a25425a2812c414336e

File: docs/sources/developers/plugins/migration-guide.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ title: Plugin migration guide
 
 ## Introduction
 
-This guide helps you identify the steps you need to take based on the Grafana version your plugin supports and explains how to migrate the plugin to the 8.2.x or a later version.
+The guide helps you identify the steps required to update a plugin from the Grafana version it currently supports to newer versions of Grafana.
```
Comment (jackw) [inline docs/sources/developers/plugins/migration-guide.md:11]:
```suggestion
This guide helps you identify the steps required to update a plugin from the Grafana version it currently supports to newer versions of Grafana.
```

Sorry I introduced a typo. 🤦

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**